lib/utils/checkAgainstRule.js

```diff
@@ -2,40 +2,33 @@
 
 const normalizeRuleSettings = require('../normalizeRuleSettings');
 const Result = require('postcss/lib/result');
-const rules = require('../rules');
+const { isPlainObject } = require('./validateTypes');
+const getStylelintRule = require('./getStylelintRule');
 
 /**
  * Useful for third-party code (e.g. plugins) to run a PostCSS Root
  * against a specific rule and do something with the warnings
- * @template T
- * @template {Object} O
- * @param {{
-		ruleName: string,
-		ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
-		root: import('postcss').Root,
-	}} options
- * @param {(warning: import('postcss').Warning) => void} callback
- * @returns {void}
+ *
+ * @type {typeof import('stylelint').utils.checkAgainstRule}
  */
 function checkAgainstRule(options, callback) {
-	if (!options)
-		throw new Error(
-			"checkAgainstRule requires an options object with 'ruleName', 'ruleSettings', and 'root' properties",
-		);
+	if (!isPlainObject(options)) throw new Error('Expected an options object');
 
-	if (!callback) throw new Error('checkAgainstRule requires a callback');
+	if (!callback) throw new Error('Expected a callback function');
 
-	if (!options.ruleName) throw new Error("checkAgainstRule requires a 'ruleName' option");
+	const { ruleName, ruleSettings, root, result } = options;
 
-	const rule = rules[options.ruleName];
+	if (!ruleName) throw new Error('Expected a "ruleName" option');
 
-	if (!rule) throw new Error(`Rule '${options.ruleName}' does not exist`);
+	const rule = getStylelintRule(ruleName, result && result.stylelint.config);
 
-	if (!options.ruleSettings) throw new Error("checkAgainstRule requires a 'ruleSettings' option");
+	if (!rule) throw new Error(`Rule '${ruleName}' does not exist`);
```

Comment:
[trivial] Could you please also convert `'` to `"` here?

```suggestion
	if (!rule) throw new Error(`Rule "${ruleName}" does not exist`);
```

Reply:
Good catch 👍 https://github.com/stylelint/stylelint/pull/6460/commits/6a53fc709c4847a360b66dcce429dd6808a5fb51